spirit/comment/tests.py

```diff
@@ -456,6 +456,45 @@ def test_comment_image_upload_invalid(self):
         self.assertIn('error', res.keys())
         self.assertIn('image', res['error'].keys())
 
+    @override_settings(MEDIA_ROOT=os.path.join(settings.BASE_DIR, 'media_test'))
+    def test_comment_file_upload(self):
+        """
+        comment file upload
+        """
+        utils.login(self)
+        file = BytesIO(b'PDFa\x01\x00\x01\x00\x80\x01\x00\x00\x00\x00ccc,\x00'
+                      b'\x00\x00\x00\x01\x00\x01\x00\x00\x02\x02D\x01\x00;')
+        files = {'file': SimpleUploadedFile('file.pdf', file.read(), content_type='application/pdf'), }
+        response = self.client.post(reverse('spirit:comment:file-upload-ajax'),
+                                    HTTP_X_REQUESTED_WITH='XMLHttpRequest',
+                                    data=files)
+        res = json.loads(response.content.decode('utf-8'))
+        file_url = os.path.join(
+            settings.MEDIA_URL, 'spirit', 'files', str(self.user.pk),  "dec7cd2c076275866cb76a4f48abffbb.file.pdf"
+        ).replace("\\", "/")
+        self.assertEqual(res['url'], file_url)
+        file_path = os.path.join(
+            settings.MEDIA_ROOT, 'spirit', 'files', str(self.user.pk), "dec7cd2c076275866cb76a4f48abffbb.file.pdf"
+        )
+        self.assertTrue(os.path.isfile(file_path))
```

Comment:
Instead of this, it would be better to check the content has been written (I know I did not do it within the image test :sob: )

```python
with open(file_path, 'rb') as fh:
    file.seek(0)
    self.assertEqual(fh.read(), file.read())
```

I ask this mostly coz I don't know if `default_storage.save` is calling `.seek(0)`. If not then call it before saving.

Reply:
The code snippet above is passing during the tests.

From what I can see (outside of the test suite), `default_storage.save` seems to be calling `.seek(0)` or something similar implicitly since calling `.seek` to various positions before saving gives the same final position via `.tell()` after saving. I am not too sure if I am checking for this the right way.

Reply:
If the above code snippet passes then the file content is being saved :+1: . I'd like to include that snippet instead of the `isfile` check, though.

Reply:
Gotcha, thanks!